PLANET-4996: Install development tools inside the containers

pablocubico opened the pull request:
See also: 
https://github.com/greenpeace/planet4-base-fork/pull/66

Added a command to install NPM and update NodeJS, but running watch inside docker is making my CPU go crazy, as said in the other PR...

comzeradd commented on Makefile:
```diff
@@ -409,6 +404,19 @@ flush:
+	# Install NPM
+	docker-compose exec php-fpm apt install npm -y
+	# Update Node version
+	docker-compose exec php-fpm sh -c 'npm cache clean -f && npm install -g n && n stable'
```
Is the plan to add `installnpm` to the `dev` target? In that case we need to adjust the [repos.sh](repos script) to use the npm from the container for each repo, right?

comzeradd commented:
Don't forget that we need to also cover `composer` and php dependencies here. In case this helps this is the system requirements that we ask people to install locally:
https://support.greenpeace.org/planet4/development/installation#system-requirements

pablocubico commented:
@comzeradd This was updated along with the base fork branch, everything works locally, but I'm not sure about the production side, the latest changes are:

- Created an `installnpm` task to install NPM on the `php-fpm` container, and made it part of the `dev` task, as I assume this is already installed in production.
- Added two commands to build and watch the assets: `assets` and `watch` (as `build` already exists)
- Added three scripts in the `base-fork` repo in a new `tasks` subfolder called `other`:
  - `build-assets.sh`: builds the assets
  - `watch.sh`: watches for changes
  - `installdeps.sh`: installs composer and npm dependencies

Any advice/suggestions welcome.

pablocubico commented on docker-compose.yml:
```diff
@@ -61,7 +61,7 @@ services:
       - APP_HOSTPATH=${APP_HOSTPATH:-}
       # - DELETE_EXISTING_FILES=false
-      - GIT_REF=${GIT_REF:-develop}
+      - GIT_REF=${GIT_REF:-planet-4996}
```
WIP: This will be restored to its original value.
